server/pulp/server/controllers/repository.py

```diff
@@ -1163,6 +1170,13 @@ def check_publish(repo_obj, dist_id, dist_inst, transfer_repo, conduit, call_con
             repo_id=repo_obj.repo_id,
             distributor_id=dist_id).update(set__last_override_config=config_override)
 
+    if force_full:
+        _logger.info(_("Fully republishing due to use of force_full in config"))
```

Comment:
I suggest to log only this case (when force_full is specified in config/override_config).
Other cases (e.g. when configs changed) don't enforce full sync/publish necessarily, plugin still can apply its optimizations and perform incremental publish or make sync almost no-op. When force_full is specified explicitly, plugin is expected to respect that.